crates/bevy_math/src/ops.rs

```diff
@@ -444,12 +447,149 @@ mod libm_ops {
+    ///
+    /// Precision is specified when the `libm` feature is enabled.
+    #[inline(always)]
+    pub fn round(x: f32) -> f32 {
+        libm::roundf(x)
+    }
+
+    /// Returns the largest integer less than or equal to `x`.
+    ///
+    /// Precision is specified when the `libm` feature is enabled.
+    #[inline(always)]
+    pub fn floor(x: f32) -> f32 {
+        libm::floorf(x)
+    }
+
+    /// Returns the fractional part of `x`.
+    ///
+    /// This function always returns the precise result.
+    #[inline(always)]
+    pub fn fract(x: f32) -> f32 {
+        libm::modff(x).0
+    }
+}
+
+#[cfg(feature = "std")]
+mod std_ops_for_no_std {
+    /// Calculates the least nonnegative remainder of `x (mod y)`.
+    ///
+    /// The result of this operation is guaranteed to be the rounded infinite-precision result.
+    #[inline(always)]
+    pub fn rem_euclid(x: f32, y: f32) -> f32 {
+        f32::rem_euclid(x, y)
+    }
+
+    /// Computes the absolute value of x.
+    ///
+    /// This function always returns the precise result.
+    #[inline(always)]
+    pub fn abs(x: f32) -> f32 {
+        f32::abs(x)
+    }
+
+    /// Returns the square root of a number.
+    ///
+    /// The result of this operation is guaranteed to be the rounded infinite-precision result.
+    /// It is specified by IEEE 754 as `squareRoot` and guaranteed not to change.
+    #[inline(always)]
+    pub fn sqrt(x: f32) -> f32 {
+        f32::sqrt(x)
+    }
+
+    /// Returns a number composed of the magnitude of `x` and the sign of `y`.
+    ///
+    /// Equal to `x` if the sign of `x` and `y` are the same, otherwise equal to `-x`. If `x` is a
+    /// `NaN`, then a `NaN` with the sign bit of `y` is returned. Note, however, that conserving the
+    /// sign bit on `NaN` across arithmetical operations is not generally guaranteed.
+    #[inline(always)]
+    pub fn copysign(x: f32, y: f32) -> f32 {
+        f32::copysign(x, y)
+    }
+
+    /// Returns the nearest integer to `x`. If a value is half-way between two integers, round away from `0.0`.
+    ///
+    /// This function always returns the precise result.
+    #[inline(always)]
+    pub fn round(x: f32) -> f32 {
+        f32::round(x)
+    }
+
+    /// Returns the largest integer less than or equal to `x`.
+    ///
+    /// This function always returns the precise result.
+    #[inline(always)]
+    pub fn floor(x: f32) -> f32 {
+        f32::floor(x)
+    }
+
+    /// Returns the fractional part of `x`.
+    ///
+    /// This function always returns the precise result.
+    #[inline(always)]
+    pub fn fract(x: f32) -> f32 {
+        f32::fract(x)
+    }
+}
+
 #[cfg(feature = "libm")]
 pub use libm_ops::*;
 
 #[cfg(not(feature = "libm"))]
 pub use std_ops::*;
 
+#[cfg(feature = "std")]
+pub use std_ops_for_no_std::*;
+
+#[cfg(all(feature = "libm", not(feature = "std")))]
+pub use libm_ops_for_no_std::*;
```

Comment:
Yeah I'm not a fan of the name but I'm not sure what else to call them haha! I've added some documentation to help explain. The idea is `std` provides _some_ of the maths operations rather than `core`, so we need to provide a compatibility shim for those operations.

In place of a better name, I've added some documentation to these new modules explaining the issue explicitly.